build/commands/scripts/mcp-setup.js

```diff
@@ -0,0 +1,415 @@
+#!/usr/bin/env node
+
+const fs = require('fs');
+const path = require('path');
+const { execSync, spawn } = require('child_process');
+const chalk = require('chalk');
+
+const isWindows = process.platform === 'win32';
+const isMacOS = process.platform === 'darwin';
+const isLinux = process.platform === 'linux';
+
+const PROJECT_ROOT = path.resolve(__dirname, '../../../../..');
+const BRAVE_CORE_ROOT = path.resolve(__dirname, '../../..');
+const CLAUDE_CONFIG_PATH = path.join(PROJECT_ROOT, '.claude', 'mcp_servers.json');
+const CURSOR_CONFIG_PATH = path.join(require('os').homedir(), '.cursor', 'mcp.json');
+const GLOBAL_CLAUDE_CONFIG_PATH = path.join(require('os').homedir(), '.claude', 'mcp_servers.json');
+const REQUIREMENTS_PATH = path.join(BRAVE_CORE_ROOT, 'agents', 'mcp', 'requirements.txt');
+
+function log(message, type = 'info') {
+  const colors = {
+    info: chalk.blue,
+    success: chalk.green,
+    warning: chalk.yellow,
+    error: chalk.red
+  };
+  console.log(colors[type](`[MCP Setup] ${message}`));
+}
+
+function ensureDirectoryExists(dirPath) {
+  if (!fs.existsSync(dirPath)) {
+    fs.mkdirSync(dirPath, { recursive: true });
+    log(`Created directory: ${dirPath}`, 'success');
+  }
+}
+
+function getPythonCommand() {
+  // Try to find the best Python command for the platform
+  const pythonCommands = ['python3', 'python'];
+  
+  for (const cmd of pythonCommands) {
+    try {
+      execSync(`${cmd} --version`, { stdio: 'pipe' });
+      return cmd;
+    } catch (e) {
+      continue;
+    }
+  }
+  throw new Error('Python not found. Please install Python 3.7+ and ensure it\'s in your PATH.');
+}
+
+function installMCPDependencies() {
+  log('Installing MCP dependencies...');
+  
+  const pythonCmd = getPythonCommand();
+  
+  try {
+    // Install MCP package
+    execSync(`${pythonCmd} -m pip install mcp`, { stdio: 'inherit' });
+    
+    // Install requirements if they exist
+    if (fs.existsSync(REQUIREMENTS_PATH)) {
+      execSync(`${pythonCmd} -m pip install -r "${REQUIREMENTS_PATH}"`, { stdio: 'inherit' });
+    }
+    
+    log('MCP dependencies installed successfully', 'success');
+  } catch (error) {
+    log(`Failed to install dependencies: ${error.message}`, 'error');
+    throw error;
+  }
+}
+
+function detectIDEs() {
+  const ides = [];
+  
+  // Check for Claude Code (look for .claude directory or running process)
+  const claudeUserDir = path.join(require('os').homedir(), '.claude');
+  if (fs.existsSync(claudeUserDir) || process.env.CLAUDE_CODE_API_KEY) {
+    ides.push('claude');
+  }
+  
+  // Check for Cursor (look for .cursor directory or in PATH)
+  const cursorUserDir = path.join(require('os').homedir(), '.cursor');
+  if (fs.existsSync(cursorUserDir)) {
+    ides.push('cursor');
+  }
+  
+  // Try to detect from running processes or common install locations
+  try {
```

Comment:
nit: the code complexity here is a bit complicated. We should abstract these into separate functions per platform.